.azure/linux-stack.yml

```diff
@@ -47,6 +47,6 @@ jobs:
     displayName: 'stack build --test --only-dependencies'
   - bash: |
       export PATH=/opt/cabal/bin:$PATH
-      stack test --ghc-options=-Werror  --stack-yaml=$STACK_YAML || stack test --ghc-options=-Werror --stack-yaml=$STACK_YAML --ta "--rerun" || LSP_TEST_LOG_COLOR=0 LSP_TEST_LOG_MESSAGES=true LSP_TEST_LOG_STDERR=true stack test --ghc-options=-Werror --stack-yaml=$STACK_YAML --ta "--rerun"
+      stack test --ghc-options=-Werror  --stack-yaml=$STACK_YAML --ta "--rerun" || stack test --ghc-options=-Werror --stack-yaml=$STACK_YAML --ta "--rerun" || LSP_TEST_LOG_COLOR=0 LSP_TEST_LOG_MESSAGES=true LSP_TEST_LOG_STDERR=true stack test --ghc-options=-Werror --stack-yaml=$STACK_YAML --ta "--rerun"
```

Comment:
This change for consistency? What does rerun the first time around mean?

Reply:
`tasty` only generates the needed log file if you set `--rerun` (or `--rerun-update`), and it can only filter tests in a second execution with `--rerun` if that log exists (tested locally in windows). 
It will execute the entire test suite if that log has no errors or does not exist. As the log is not saved between ci builds and that is the way you make the second test execution to handle only failing tests. As all calls are chained with `||`, if the first execution is succesful, no more executions will be done.

Reply:
Maybe use each concrete flag (`--rerun-update` in the first one, `--rerun-filter` in second and third) in each execution would make it clearer

Reply:
I'd much rather the first flag to be `--rerun-create-afresh` or similar - we do caching on the CI, so I worry about stale test success being cached. While it probably isn't, using different flags would make it clear.

Reply:
`--rerun-update` creates or replaces the log so i think it would be like `--rerun-create-afresh`